### PR TITLE
[rush] Improve the way static assets are handled.

### PR DESCRIPTION
**Pull request by iclanton**
## Summary

This PR cleans up some hardcoded relative paths and improves the way static assets are copied.

## How it was tested

Ran `rush init` and `rush update` (for the pnpmfile shim and the `common/scripts` scripts);

### CODE REVIEW

File: libraries/rush-lib/src/logic/StandardScriptUpdater.ts [hunk 1]
```diff
@@ -96,11 +166,24 @@ export class StandardScriptUpdater {
   private static _normalize(content: string): string {
     // Ignore newline differences from .gitattributes
     return (
-      Text.convertToLf(content)
```
Comment (octogonz) [inline libraries/rush-lib/src/logic/StandardScriptUpdater.ts:99]: Why did you remove `convertToLf()` here?

File: libraries/rush-lib/src/utilities/PathConstants.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See the @microsoft/rush package's LICENSE file for license information.
+
+import { PackageJsonLookup } from '@rushstack/node-core-library';
+
+/**
+ * The currently-executing rush-lib package's root folder path.
+ */
+export const rushLibFolderRootPath: string = PackageJsonLookup.instance.tryGetPackageFolderFor(__dirname)!;
```
Comment (octogonz) [inline libraries/rush-lib/src/utilities/PathConstants.ts:9]: If we substituted a hardwired `path.resolve(__dirname, '../../../')` would that be significantly more efficient than `PackageJsonLookup` here?  I'm thinking of cases such as tab-completion that eval relatively little code.

**Comment by octogonz** [comment]
Approved with some comments

**Review by octogonz — APPROVED** [review]

File: libraries/rush-lib/src/logic/StandardScriptUpdater.ts (hunk 1 above)
Comment (iclanton) [reply libraries/rush-lib/src/logic/StandardScriptUpdater.ts:99]: The `trimRight()` on each line will trim the `\r` if the file is CRLF.

File: libraries/rush-lib/src/utilities/PathConstants.ts (hunk 2 above)
Comment (iclanton) [reply libraries/rush-lib/src/utilities/PathConstants.ts:9]: Yes, but I'm going to be raising a PR that includes shipping a webpacked version of `rush-lib` in addition to the `lib` folder version, so a hardcoded path won't work.

**Comment by elliot-nelson** [comment]
Hey @iclanton FYI -- after pulling latest main, building, and attempting to run "rush update" using the local copy I am getting an error:

```console
ERROR: File does not exist: /Users/enelson/dev/rushstack2/libraries/rush-lib/lib/scripts/PnpmfileShim.js
ENOENT: no such file or directory, stat '/Users/enelson/dev/rushstack2/libraries/rush-lib/lib/scripts/PnpmfileShim.js'
```
